### PR TITLE
Carry the metrics status in RollingUpgrade CR

### PR DESCRIPTION
**Pull request by shaoxt**
1. Carry the metrics status in RollingUpgrade CR
2. Add Terminating wait in terminated step

### CODE REVIEW

**Comment by shaoxt** [comment]
```
spec:
  asgName: sheldon08-instance-manager-nodes
  nodeIntervalSeconds: 300
  postDrainScript: |
    echo "Hello, postDrain!"
  postDrainWaitScript: |
    echo "Hello, postDrainWait!"
  postTerminateScript: |
    echo "Hello, postTerminate!"
  strategy:
    drainTimeout: 900
    maxUnavailable: 1
    mode: eager
    type: randomUpdate
status:
  completePercentage: 100%
  currentStatus: completed
  endTime: "2021-06-24T05:55:36Z"
  lastDrainTime: "2021-06-24T05:50:23Z"
  lastTerminationTime: "2021-06-24T05:50:28Z"
  nodesProcessed: 3
  startTime: "2021-06-24T05:34:03Z"
  statistics:
  - durationCount: 3
    durationSum: 5.41374933s
    stepName: kickoff
  - durationCount: 3
    durationSum: 6m0.206569857s
    stepName: desired_node_ready
  - durationCount: 3
    durationSum: 150.02µs
    stepName: predrain_script
  - durationCount: 3
    durationSum: 307.092159ms
    stepName: drain
  - durationCount: 3
    durationSum: 30.176µs
    stepName: postdrain_script
  - durationCount: 3
    durationSum: 151.271µs
    stepName: post_wait
  - durationCount: 3
    durationSum: 5.542257165s
    stepName: terminate
  - durationCount: 3
    durationSum: 39.095µs
    stepName: post_terminate
  - durationCount: 3
    durationSum: 14m59.999999588s
    stepName: terminated
  - durationCount: 3
    durationSum: 21m9.056289521s
    stepName: total
  totalNodes: 3
  totalProcessingTime: 21m33s
```

File: controllers/metrics.go [hunk 1]
```diff
@@ -62,9 +68,8 @@ func (s *RollingUpgradeContext) ToStepDuration(groupName, nodeName string, stepN
 	}
 }
 
-// Node turns onto step
-func (s *RollingUpgradeContext) NodeStep(InProcessingNodes map[string]*v1alpha1.NodeInProcessing,
-	nodeSteps map[string][]v1alpha1.NodeStepDuration, groupName, nodeName string, stepName v1alpha1.RollingUpgradeStep) {
+func (s *RollingUpgradeContext) DoNodeStep(InProcessingNodes map[string]*v1alpha1.NodeInProcessing,
```
Comment (shreyas-badiger) [inline controllers/metrics.go:71]: Why not merge both NodeStep() and DoNodeStep()?  I mean, any specific reasons for why you aren't merging the DoNodeStep() into NodeStep()?

Comment (shaoxt) [reply controllers/metrics.go:71]: If anywhere needs to call DoNodeStep. They have to pass the "time.Now()"

Comment (shaoxt) [reply controllers/metrics.go:71]: DoNodeStep needs to parameterize the endTime

**Pull request merged**